### PR TITLE
[Doc] Update docs about materialized view

### PR DESCRIPTION
**Pull request by LiShuMing**
Why I'm doing:

Docs about materialized view is late than the codes.

What I'm doing:

Update docs to be better for the current implentments.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_MATERIALIZED_VIEW.md [hunk 1]
```diff
@@ -235,8 +238,9 @@ The partitioning strategy of the asynchronous materialized view. As for the curr
...
+- `date_trunc` function: The function used to truncate the time unit. `PARTITION BY date_trunc("MONTH", dt)` means that the `dt` column is truncated to month as the unit for partitioning. The `date_trunc` function supports truncating time to units including `YEAR`, `MONTH`, `DAY`, `HOUR`, and `MINUTE`.
+- `str2date` function: The function used to partition string type parititions of base table into materialized view's partition. `PARTITION BY str2date(dt, "%Y%m%d")` means that the `dt` column is a string date type whichi its date formate is `"%Y%m%d"`. The `str2date` function supports a lot of date formats, you can refer [str2date](../sql-reference/sql-functions/date-time-functions/str2date.md) function reference for more information.
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_MATERIALIZED_VIEW.md:242]: 1. spelling: whichi -- which    2. grammar: whichi its -- whose???    3. formate  --- format    4. you can refer to    5. remove "function reference"

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_MATERIALIZED_VIEW.md [hunk 2]
```diff
@@ -722,3 +800,56 @@ INNER JOIN customer AS c ON c.C_CUSTKEY = l.LO_CUSTKEY
...
+  `d_sellingseason` varchar(14) DEFAULT NULL,
+  `d_lastdayinweekfl` int(11) DEFAULT NULL,
+  `d_lastdayinmonthfl` int(11) DEFAULT NULL,
+  `d_holidayfl` int(11) DEFAULT NULL,
+  `d_weekdayfl` int(11) DEFAULT NULL,
+  `d_datekey` varchar(11) DEFAULT NULL
+) partition by (d_datekey);
+
+
+-- Create the materialied view  with `str2date`
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_MATERIALIZED_VIEW.md:830]: add a period at the end

File: docs/zh/sql-reference/sql-statements/data-definition/CREATE_MATERIALIZED_VIEW.md [hunk 3]
```diff
@@ -244,6 +248,7 @@ AS
 
 异步物化视图的属性。您可以使用 [ALTER MATERIALIZED VIEW](./ALTER_MATERIALIZED_VIEW.md) 修改已有异步物化视图的属性。
 
+- `session.`: 如果您想要更改与物化视图相关的 Session 变量属性，必须在属性前添加 `session.` 前缀，例如，`session.query_timeout`。对于非 Session 属性，例如，`mv_rewrite_staleness_second`，则无需指定前缀。
```
Comment (evelynzhaojie) [inline docs/zh/sql-reference/sql-statements/data-definition/CREATE_MATERIALIZED_VIEW.md:251]: 用户知道有哪些 session 变量可以更改吗，建议给个链接

**Review by EsoragotoSpirit — DISMISSED** [review]

**Review by amber-create — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-3.1

**Pull request merged**